store/sqlstore/preference_store.go

```diff
@@ -236,18 +236,21 @@ func (s SqlPreferenceStore) IsFeatureEnabled(feature, userId string) (bool, *mod
 	return value == "true", nil
 }
 
-func (s SqlPreferenceStore) Delete(userId, category, name string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		if _, err := s.GetMaster().Exec(
-			`DELETE FROM
-				Preferences
-			WHERE
-				UserId = :UserId
-				AND Category = :Category
-				AND Name = :Name`, map[string]interface{}{"UserId": userId, "Category": category, "Name": name}); err != nil {
-			result.Err = model.NewAppError("SqlPreferenceStore.Delete", "store.sql_preference.delete.app_error", nil, err.Error(), http.StatusInternalServerError)
-		}
-	})
+func (s SqlPreferenceStore) Delete(userId, category, name string) *model.AppError {
+	query :=
+		`DELETE FROM Preferences
+		WHERE
+			UserId = :UserId
+			AND Category = :Category
+			AND Name = :Name`
+
+	_, err := s.GetMaster().Exec(query, map[string]interface{}{"UserId": userId, "Category": category, "Name": name})
+
```

Comment:
Nit: You can remove this line.

Reply:
Isn't that basically the main purpose of the function?

Reply:
You mean `Name`?

Reply:
I'm talking about the empty line L248.

Reply:
Ohh ok. Thank you!